File: rc6algorithmangularnetcore.client/src/app/models/Invoice.ts
export interface Invoice {
    id?: number;
    companyId?: number;
    invoiceNumber?: string;
    customerName?: string;
    customerTaxNumber?: string;
    customerAddress?: string;
    issueDate?: string;
    paymentDate?: string;
    fulfilmentDate?: string;
    netValue?: number;
    vat?: 8 | 16 | 27;
    grossValue?: number;
    currency?: string;
}

export class InvoiceDTO implements Invoice { }

File: RC6AlgorithmAngularNetCore.Server/Entities/Invoice.cs
namespace RC6AlgorithmAngularNetCore.Server.Entities
{
    public class Invoice
    {
        public int Id { get; set; }
        public int CompanyId { get; set; }

        public required string InvoiceNumber { get; set; }

        public string CustomerName { get; set; }
        public string CustomerTaxNumber { get; set; }
        public string CustomerAddress { get; set; }
        public required string IssueDate { get; set; }
        public string PaymentDate { get; set; }

        public string FulfilmentDate { get; set; }

        public required float NetValue { get; set; }
        public float GrossValue { get; set; }
        public float VAT { get; set; }
        public string Currency { get; set; }
    }
}
